src/Microsoft.Diagnostics.DebugServices.Implementation/MetadataMappingMemoryService.cs

```diff
@@ -20,26 +20,39 @@ namespace Microsoft.Diagnostics.DebugServices.Implementation
     /// that older (less than 5.0) createdumps generate  so it needs this special 
     /// metadata  mapping memory service.
     /// </summary>
-    public class MetadataMappingMemoryService : IMemoryService
+    public class MetadataMappingMemoryService : IMemoryService, IDisposable
     {
-        private readonly ITarget _target;
+        private readonly IServiceContainer _container;
         private readonly IMemoryService _memoryService;
+        private readonly IRuntimeService _runtimeService;
+        private readonly ISymbolService _symbolService;
         private bool _regionInitialized;
         private ImmutableArray<MetadataRegion> _regions;
-        private IRuntimeService _runtimeService;
-        private ISymbolService _symbolService;
 
         /// <summary>
         /// Memory service constructor
         /// </summary>
-        /// <param name="target">target instance</param>
+        /// <param name="container">service container to clone</param>
         /// <param name="memoryService">memory service to wrap</param>
-        public MetadataMappingMemoryService(ITarget target, IMemoryService memoryService)
+        public MetadataMappingMemoryService(IServiceContainer container, IMemoryService memoryService)
         {
-            _target = target;
+            _container = container.Clone();
```

Comment:
Any reason not to clone it in the parent? That way in general this one can't modify the other one.

Reply:
I wanted the code that needs the cloning actually do it.

Reply:
Some one needs to dispose the service instances too for the clone which it the life time of the mapping memory service